core/src/main/java/tech/tablesaw/io/FileReader.java

```diff
@@ -25,42 +27,67 @@
   private static Logger logger = LoggerFactory.getLogger(FileReader.class);
   private static final int UNLIMITED_SAMPLE_SIZE = -1;
 
+  /**
+   * @deprecated Use {@link #getColumnTypes(Reader, ReadOptions, int, AbstractParser, String[])} }
+   */
+  @Deprecated
+  public ColumnType[] getColumnTypes(
+      Reader reader, ReadOptions options, int linesToSkip, AbstractParser<?> parser) {
+    return getColumnTypes(reader, options, linesToSkip, parser, null);
+  }
```

Comment:
I can deprecate, but I think that this should be deprecated as it continues only to maintain binary compatibility as this is a public method.
In fact, I don't know why this method is public, but the correct one to call by now should be the variant with columnNames, because is the one that auto column detection logic will use.
If you call the old one you can get a different result than the ones you get by autodetecting column types.

The ones I remove deprecation were the columnType ones that you pointed in the first review.

Reply:
Taking a second thought, if it's important to maintain that interface, I can remove the deprecated and change the current implementation making it call the `getColumnNames` method for the columnNames param that now is passed null.